### PR TITLE
Added a test to cover the featurization of reductions

### PR DESCRIPTION
**Pull request by benoitsteiner**

### CODE REVIEW

File: apps/autoscheduler/test_function_dag.cpp [hunk 1]
```diff
@@ -47,11 +44,11 @@ int main(int argc, char **argv) {
         std::vector<Var> vars = {x, y};
         Halide::Type input_type = Halide::Float(32);
         g.define_extern(
-            "generate_output_vals",
+            "mul_by_two",
             {arg},
             input_type,
             vars,
-            Halide::NameMangling::C);
+            Halide::NameMangling::C /*PlusPlus*/);
```
Comment (steven-johnson) [inline apps/autoscheduler/test_function_dag.cpp]: Why is PlusPlus commented out?

File: apps/autoscheduler/test_function_dag.cpp [hunk 2]
```diff
@@ -81,6 +78,102 @@ int main(int argc, char **argv) {
 
     // Disabled for now: there is still work to do to populate the jacobian
     //assert(with_extern.str() == without_extern.str());
+}
+
+extern "C" int matmul(
+    halide_buffer_t *input1,
+    halide_buffer_t *input2,
+    halide_buffer_t *output) {
+
```
Comment (steven-johnson) [inline apps/autoscheduler/test_function_dag.cpp]: Nit: first line of function should not be blank

File: apps/autoscheduler/test_function_dag.cpp [hunk 3]
```diff
@@ -81,6 +78,102 @@ int main(int argc, char **argv) {
...
+                (*out) += (*in1) * (*in2);
+            }
+        }
+    }
+    return 0;
+}
+
+void test_matmul(const MachineParams &params, const Target &target) {
+
```
Comment (steven-johnson) [inline apps/autoscheduler/test_function_dag.cpp]: ditto

File: apps/autoscheduler/test_function_dag.cpp (hunk 1 above)
Comment (benoitsteiner) [reply apps/autoscheduler/test_function_dag.cpp]: It's a mistake, fixed :)

File: apps/autoscheduler/test_function_dag.cpp (hunk 3 above)
Comment (benoitsteiner) [reply apps/autoscheduler/test_function_dag.cpp]: done

File: apps/autoscheduler/test_function_dag.cpp (hunk 2 above)
Comment (benoitsteiner) [reply apps/autoscheduler/test_function_dag.cpp]: done

**Pull request merged**